### PR TITLE
Support for Batch Request Guild Members

### PR DESCRIPTION
**Pull request by mistahj67**
This is supported by the Discord API and will save requests.

### CODE REVIEW

**Comment by Bios-Marcel** [comment]
This is backwards compatibility breaking change. I'd suggest adding a new function.

**Comment by mistahj67** [comment]
Updated to eliminate breaking issue

File: wsapi.go [hunk 1]
```diff
@@ -399,9 +399,10 @@ func (s *Session) UpdateStatusComplex(usd UpdateStatusData) (err error) {
...
 
 type requestGuildMembersData struct {
-	GuildID []string `json:"guild_id"`
-	Query   string   `json:"query"`
-	Limit   int      `json:"limit"`
+	GuildID  string   `json:"guild_id"`
+	GuildIDs []string `json:"guild_id"`
+	Query    string   `json:"query"`
+	Limit    int      `json:"limit"`
 }
```
Comment (JurrijnP) [inline wsapi.go]: You can leave the struct the same as before so:
```go
type requestGuildMembersData struct {
	GuildID []string `json:"guild_id"`
	Query   string   `json:"query"`
	Limit   int      `json:"limit"`
}
```

And then pass the argument in the original function to the slice.

File: wsapi.go [hunk 2]
```diff
@@ -411,10 +411,35 @@ type requestGuildMembersOp struct {
...
+	data := requestGuildMembersData{
+		GuildIDs: guildIDs,
+		Query:    query,
+		Limit:    limit,
+	}
+	err = s.requestGuildMember(data)
+	return err
+}
+
+func (s *Session) requestGuildMember(data requestGuildMembersData) (err error) {
```
Comment (CarsonHoffman) [inline wsapi.go]: `requestGuildMember` seems to be a typo here; `requestGuildMembers` would make more sense.

File: wsapi.go [hunk 3]
```diff
@@ -423,17 +448,11 @@ func (s *Session) RequestGuildMembers(guildID, query string, limit int) (err err
...
-		Query:   query,
-		Limit:   limit,
-	}
-
 	s.wsMutex.Lock()
 	err = s.wsConn.WriteJSON(requestGuildMembersOp{8, data})
 	s.wsMutex.Unlock()
 
-	return
+	return err
```
Comment (CarsonHoffman) [inline wsapi.go]: This is unnecessary as `err` is assigned on line 452.

Comment (mistahj67) [reply wsapi.go]: It's required to return err as the function returns an error?

Comment (CarsonHoffman) [reply wsapi.go]: Since `err` is a named return value, it's redundant. See: [named return values](https://tour.golang.org/basics/7)

Comment (mistahj67) [reply wsapi.go]: TIL. Thanks.

**Pull request merged**